src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -644,8 +644,12 @@ protected virtual Saml2Conditions CreateConditions(SecurityTokenDescriptor token
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.AudienceRestrictions.Add(new Saml2AudienceRestriction(tokenDescriptor.Audience));
+            var audienceRestriction = new Saml2AudienceRestriction(tokenDescriptor.Audiences);
```

Comment:
What happens if Audiences is null?

Reply:
I don't think it can be null, in the SecurityTokenDescriptor class, we use Interlocked.CompareExchange to initialize it to a zero length list on first access.

Reply:
This is correct, lazy instantiation ensures not null which was a great idea on your part @brentschmaltz saved me from a lot of null checks :)

Reply:
I agree, so check the count, if the count is 0, why call the ctor, then call again to add the tokenDescriptor.Audience.

Reply:
The only way around this without more calls to the Saml2AudienceRestriction object was dropping the local variable entirely and having a little bit of duplicate code, so I did that.